### PR TITLE
SE UTs: Replace ValidateTag with TagValue

### PR DESCRIPTION
**Pull request by pavel-mikula-sonarsource**
`ValidatorTestCheck.ValidateTag` is now obsolete, because we can fluently assert symbolic value constraints.

Using `TagValue("Name")` also gives better error message when assertion fails, because we can see what tagname failed in the message.

Before:
```
validator.ValidateTag("If", x => x.Should().HaveOnlyConstraint(ObjectConstraint.Null));
```

After:
```
 validator.TagValue("If").Should().HaveOnlyConstraint(ObjectConstraint.Null);
 ```

All usages were replaced using regex:
Find: `ValidateTag\((?<Name>"[^"]+"), x => x.Should\(\)\.(?<Action>.*)\);`
Replace: `TagValue(${Name}).Should().${Action};`

### CODE REVIEW

**Review by antonioaversa — CHANGES_REQUESTED** [review]
The change has not been applied to all invocations of `ValidateTag` which don't conform with the regex.

So, many occurrences of `ValidateTag` still persist in the code base.

Running a file search in VS at solution level on the string ".ValidateTag(" (no Match case, no Match whole word, no Use regular expressions) reveals many occurrences remaining in:
- C:\dev\sonar-dotnet\analyzers\tests\SonarAnalyzer.UnitTest\SymbolicExecution\Roslyn\RoslynSymbolicExecutionTest.cs
- C:\dev\sonar-dotnet\analyzers\tests\SonarAnalyzer.UnitTest\SymbolicExecution\Roslyn\RoslynSymbolicExecutionTest.*.cs

Examples of UT which still contain deprecated calls:
- `BinaryEqualsNull_SetsBoolConstraint_KnownResult_CS`,
- `BinaryEqualsNull_SetsBoolConstraint_KnownResult_VB`,
- `BinaryNotEqualsNull_SetsBoolConstraint_KnownResult_CS`,
- `BinaryNotEqualsNull_SetsBoolConstraint_KnownResult_VB`,
- `Binary_NullableRelationalNonNull_SetsObjectConstraint_CS`,

The change for HasConstraint seems easy enough, and could be automatized with another regex.
Basically:
- something like `validator.ValidateTag("IsTrue", x => x.HasConstraint(BoolConstraint.True).Should().BeTrue());` 
- should be replaced by `validator.TagValue("IsTrue").HasConstraint(BoolConstraint.True).Should().BeTrue();`.

I would try something like (barely tested on a single example):
- something like `ValidateTag\((?<Name>"[^"]+"), x => x\.HasConstraint(?<Action>.*)\);`
- should be replaced by `validator.ValidateTag("IsTrue", x => x.HasConstraint(BoolConstraint.True).Should().BeTrue());`

**Comment by pavel-mikula-sonarsource** [comment]
These kinds of assertions need to be migrated to the relatively new `Should().HaveOnlyConstraints(...)`, and that's another task. That's why I left the method there and didn't delete it yet.
```
validator.ValidateTag("IsTrue", x => x.HasConstraint(BoolConstraint.True).Should().BeTrue());
```

**Review by antonioaversa — APPROVED** [review]
LGTM
